test/hotspot/jtreg/runtime/Monitor/DeflationIntervalsTest.java

```diff
@@ -0,0 +1,357 @@
+ */
+
+/*
+ * @test id=noGuaranteedGADT_threshold_safepoint
+ * @requires vm.flagless
+ * @library /test/lib
+ * @run driver DeflationIntervalsTest noGuaranteedGADT_threshold_safepoint
+ */
+
+/*
+ * @test id=guaranteed_threshold_noSafepoint
+ * @requires vm.flagless
+ * @library /test/lib
+ * @run driver DeflationIntervalsTest guaranteed_threshold_noSafepoint
+ */
+
+/*
+ * @test id=guaranteed_threshold_safepoint
+ * @requires vm.flagless
+ * @library /test/lib
+ * @run driver DeflationIntervalsTest guaranteed_threshold_safepoint
+ */
+
+public class DeflationIntervalsTest {
+
+    public static class Test {
+        // Inflate a lot of monitors, so that threshold heuristics definitely fires
+        private static final int MONITORS = 10_000;
+
+        // Use a handful of threads to inflate the monitors, to eat the cost of
+        // wait(1) calls. This can be larger than available parallelism, since threads
+        // would be time-waiting.
+        private static final int THREADS = 16;
+
+        private static Thread[] threads;
+        private static Object[] monitors;
+
+        public static void main(String... args) throws Exception {
+            monitors = new Object[MONITORS];
+            threads = new Thread[THREADS];
+
+            for (int t = 0; t < THREADS; t++) {
+                int monStart = t * MONITORS / THREADS;
+                int monEnd = (t + 1) * MONITORS / THREADS;
+                threads[t] = new Thread(() -> {
+                    for (int m = monStart; m < monEnd; m++) {
+                        Object o = new Object();
+                        synchronized (o) {
+                            try {
+                                o.wait(1);
+                            } catch (InterruptedException e) {
+                            }
+                        }
+                        monitors[m] = o;
+                    }
+                });
+                threads[t].start();
+            }
+
+            for (Thread t : threads) {
+                t.join();
+            }
+
+            try {
+                Thread.sleep(10_000);
+            } catch (InterruptedException ie) {
+            }
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        if (args.length < 1) {
+            throw new IllegalArgumentException("Expect the test label");
+        }
+
+        String test = args[0];
+        switch (test) {
+            case "defaults":
+                // Try with all defaults
+                test(Disabled.NO, Guaranteed.MAYBE, Threshold.MAYBE);
+                break;
+
+            case "allIntervalsZero":
+                // Try with all deflation intervals at zero
+                test(Disabled.YES, Guaranteed.NO, Threshold.NO,
+                    "-XX:GuaranteedAsyncDeflationInterval=0",
+                    "-XX:AsyncDeflationInterval=0",
+                    "-XX:MonitorUsedDeflationThreshold=0",
```

Comment:
But... this one isn't an interval option...

Reply:
Right. Dropped! Has no effect on the test outcome.